tests/suites/test_suite_psa_crypto.function

```diff
@@ -2007,6 +2007,43 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void raw_agreement_key_policy( int policy_usage,
+                               int policy_alg,
+                               int key_type_arg,
+                               data_t *key_data,
+                               int exercise_alg )
+{
+    psa_key_handle_t handle = 0;
+    psa_key_policy_t policy = PSA_KEY_POLICY_INIT;
+    psa_key_type_t key_type = key_type_arg;
+    psa_crypto_generator_t generator = PSA_CRYPTO_GENERATOR_INIT;
+    psa_status_t status;
+
+    PSA_ASSERT( psa_crypto_init( ) );
+
+    PSA_ASSERT( psa_allocate_key( &handle ) );
+    psa_key_policy_set_usage( &policy, policy_usage, policy_alg );
+    PSA_ASSERT( psa_set_key_policy( handle, &policy ) );
+
+    PSA_ASSERT( psa_import_key( handle, key_type,
+                                key_data->x, key_data->len ) );
+
+    status = raw_key_agreement_with_self( exercise_alg, handle );
+
+    if( policy_alg == exercise_alg &&
+        ( policy_usage & PSA_KEY_USAGE_DERIVE ) != 0 )
+        PSA_ASSERT( status );
+    else
+        TEST_EQUAL( status, PSA_ERROR_NOT_PERMITTED );
```

Comment:
Do we always expect `PSA_ERROR_NOT_PERMITTED` when `policy_alg != exercise_alg`?

Do you prefer this extra logic in the test or would you rather add an expected status parameter? I personally find an expected status parameter easier to read and when writing tests, easier to know I'm testing what I want to test (less likely to work on accident).

Reply:
I prefer to have the logic inside the test code. The more logic you put in test data, the bigger the risk of copypasta. It's also more readable that way, as logic rather than as a bunch of special cases.